### PR TITLE
ARTEMIS-3620 Journal blocking delete/update record with no sync

### PR DESCRIPTION
**Pull request by franz1981**
https://issues.apache.org/jira/browse/ARTEMIS-3620

### CODE REVIEW

File: artemis-journal/src/main/java/org/apache/activemq/artemis/core/journal/impl/JournalImpl.java [hunk 1]
```diff
@@ -1161,20 +1162,27 @@ public void run() {
...
+      } else {
+         onFoundAddInfo = new SimpleFutureImpl<>();
+      }
+      if (onFoundAddInfo == null) {
+         internalAppendDeleteRecord(id, false, null, callback);
+      } else {
+         internalAppendDeleteRecord(id, sync, (record, result) -> onFoundAddInfo.set(result), callback);
+      }
+      if (onFoundAddInfo != null) {
+         if (!onFoundAddInfo.get()) {
```
Comment (clebertsuconic) [inline artemis-journal/src/main/java/org/apache/activemq/artemis/core/journal/impl/JournalImpl.java:1182]: I'm thinking we should just stop doing that.. just always log, like we do on updates...

The only outcome from the IllegalStateException would be log.warn in most cases.. especially on delete.

**Comment by franz1981** [comment]
Some results of this change.
Pre:
```
Benchmark                                          (compactMinFiles)  (compactPercentage)  (fileSize)  (minFiles)  (poolSize)  (recordSize)  (records)  Mode  Cnt      Score     Error  Units
JournalSyncDeletePerfTest.batchAppendUpdateDelete                 10                   30    10485760           4          20            64        100  avgt   20   1476.175 ±  26.177  us/op
JournalSyncDeletePerfTest.batchAppendUpdateDelete                 10                   30    10485760           4          20            64       1000  avgt   20  14994.321 ± 590.080  us/op
```
Post:
```
Benchmark                                    (compactMinFiles)  (compactPercentage)  (fileSize)  (minFiles)  (poolSize)  (recordSize)  (records)  Mode  Cnt     Score     Error  Units
JournalSyncDeletePerfTest.batchAppendDelete                 10                   30    10485760           4          20            64        100  avgt   20   489.463 ±  28.756  us/op
JournalSyncDeletePerfTest.batchAppendDelete                 10                   30    10485760           4          20            64       1000  avgt   20  4476.861 ± 139.642  us/op
```

**Comment by franz1981** [comment]
CI is green with it so I can move it into non-draft ;)

File: artemis-journal/src/main/java/org/apache/activemq/artemis/core/journal/impl/JournalImpl.java [hunk 2]
```diff
@@ -1161,20 +1168,25 @@ public void run() {
...
+         onFoundAddInfo = null;
+      } else {
+         onFoundAddInfo = new SimpleFutureImpl<>();
+      }
+      if (onFoundAddInfo == null) {
+         internalAppendDeleteRecord(id, false, null, callback);
+      } else {
+         internalAppendDeleteRecord(id, sync, (record, result) -> onFoundAddInfo.set(result), callback);
+      }
+      if (onFoundAddInfo != null && !onFoundAddInfo.get()) {
```
Comment (gemmellr) [inline artemis-journal/src/main/java/org/apache/activemq/artemis/core/journal/impl/JournalImpl.java:1187]: Similar to earlier comment. Did we start being charged for newlines and noone told me? :)

**Comment by franz1981** [comment]
@clebertsuconic in order to help other prj that relies on these fixes I plan to merge it today if none complains (after fixing what @gemmellr suggested)

**Comment by clebertsuconic** [comment]
@franz1981 is this ready to merge? there was some aesthetic suggestions by Robbie.

**Pull request closed without merging**